Userland/tail.cpp

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <getopt.h>
+#include <unistd.h>
+#include <AK/Assertions.h>
+#include <LibCore/CFile.h>
+
+static void print_usage_and_exit()
+{
+    printf("usage: tail [-f, -n <linecount>] <filename>\n");
+    exit(1);
+}
+
+void tail_from_pos(CFile& file, off_t startline, bool want_follow)
+{
+    if (!file.seek(startline + 1))
+        return;
+
+    while (true) {
+        const auto& b = file.read(4096);
+        if (b.is_empty()) {
+            if (!want_follow) {
+                break;
+            } else {
+                while (!file.can_read()) {
+                    // FIXME: would be nice to have access to can_read_from_fd with an infinite timeout
+                    usleep(100);
+                }
+                continue;
+            }
+        }
+
+        write(STDOUT_FILENO, b.pointer(), b.size());
```

Comment:
We need to check for errors here and bail. If we are part of a pipe chain, writing to stdout may fail.